great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -327,14 +371,25 @@ def reconcile_profiler_variables(
         on key "miss", and replaces the original on key "hit" (or "collision"), because "variables" is a unique member.
 
         :param variables: variables overrides, supplied in dictionary (configuration) form
+        :param reconciliation_strategy: one of update, nested_update, or overwrite ways of reconciling overwrites
         :return: reconciled variables in their canonical ParameterContainer object form
         """
         effective_variables: ParameterContainer
         if variables is not None and isinstance(variables, dict):
             variables_configs: dict = self.variables.to_dict()["parameter_nodes"][
                 "variables"
             ]["variables"]
-            variables_configs.update(variables)
+
+            if reconciliation_strategy == ReconciliationStrategy.NESTED_UPDATE:
+                variables_configs = nested_update(
+                    variables_configs,
+                    variables,
+                )
+            elif reconciliation_strategy == ReconciliationStrategy.REPLACE:
+                variables_configs = variables
+            elif reconciliation_strategy == ReconciliationStrategy.UPDATE:
+                variables_configs.update(variables)
```

Comment:
Does it make sense to wrap this logic in a method? Either as a helper within RBP or more likely in the actual `ReconciliationStrategy` dataclass? We seem to do this conditional logic more than once.

I see that the logic is a bit more complicated in a few spots so maybe it makes sense to keep things verbose.

Reply:
@cdkini I made this verbose, because these strategy settings are not all applied together.  For instance, `variable` is at one level (Profiler), while the rest are at another level (Rule).  I would prefer to keep it like this for now as it makes it easier to "program" these settings.  Thanks.